charmcraft/commands/build.py

```diff
@@ -440,6 +444,27 @@ def handle_dispatcher(self, linked_entrypoint):
                 relative_link = relativise(dest_hook, dispatch_path)
                 dest_hook.symlink_to(relative_link)
 
+    def handle_schemas(self):
+        """Handle fetching any defined schemas."""
+        logger.debug("Fetching any schemas from metadata.yaml")
+        metadata_yaml = self.charmdir / CHARM_METADATA
+
+        metadata = yaml.safe_load(metadata_yaml.read_text())
+
+        for rel_type in ("requires", "provides", "peers"):
+            for relation in metadata.get(rel_type, {}).values():
+                if "schema" in relation and isinstance(relation["schema"], str):
+                    schema = relation["schema"]
+                    try:
+                        response = requests.get(schema)
+                        response.raise_for_status()
+                    except requests.RequestException:
+                        raise CommandError(f"Couldn't fetch schema `{schema}`")
+
+                    relation["schema"] = yaml.safe_load(response.text)
+
+        metadata_yaml.write_text(yaml.safe_dump(metadata), encoding="utf8")
```

Comment:
This approach to updating the YAML will result in the loss of comments.  It will also re-write the file in-place.  Can the schemas be stored separately?

Reply:
Good point. The SDI library is doing this for reasons that aren't relevant with `charmcraft` supporting this itself. Updated to leave metadata alone and store the schemas in the built charm file under a `schemas` directory, let me know if that works for you.